basic_autonomy/src/basic_autonomy.cpp

```diff
@@ -234,6 +234,198 @@ namespace basic_autonomy
             return constrained_points;
         }
 
+        std::vector<lanelet::BasicPoint2d> create_lanechange_route(int ending_lane_id, double starting_downtrack, double ending_downtrack,
+                                                                   const carma_wm::WorldModelConstPtr &wm,const cav_msgs::VehicleState &state)
+        {
+            std::vector<lanelet::BasicPoint2d> centerline_points = {};
+
+            //Get starting lanelet and ending lanelets
+            auto lanelets = wm->getLaneletsBetween(starting_downtrack, ending_downtrack,true);
```

Comment:
Should we still be using the starting lanelet id from the maneuver if the tactical plugin is being asked to plan before it?
We do update the starting downtrack to be the current downtrack in case we are behind. Would that still be valid?

Reply:
I think if the maneuver's starting lanelet does not include the starting downtrack than it is an invalid maneuver.